vm/vm.go

```diff
@@ -127,13 +150,20 @@ func Execute(txns []core.Transaction, declaredClasses []core.Class, blockNumber,
 
 	txnsJSON, classesJSON, err := marshalTxnsAndDeclaredClasses(txns, declaredClasses)
 	if err != nil {
-		return nil, err
+		return nil, nil, err
 	}
 
 	txnsJSONCstr := C.CString(string(txnsJSON))
 	classesJSONCStr := C.CString(string(classesJSON))
 
-	sequencerAddressBytes := sequencerAddress.Bytes()
+	var address *felt.Felt
+	if sequencerAddress != nil {
+		address = sequencerAddress
+	} else {
+		address = core.NetworkBlockHashMetaInfo(network).FallBackSequencerAddress
+	}
+	sequencerAddressBytes := address.Bytes()
+
```

Comment:
can we move this to caller site? it feels like it doesnt belong here

Reply:
Moved